packages/@react-aria/table/src/useTableColumnResize.ts

```diff
@@ -59,13 +59,16 @@ export function useTableColumnResize<T>(props: AriaTableColumnResizeProps<T>, st
       columnResizeWidthRef.current = stateRef.current.getColumnWidth(item.key);
       cursor.current = document.body.style.cursor;
     },
-    onMove({deltaX, pointerType}) {
+    onMove({deltaX, deltaY, pointerType}) {
       if (direction === 'rtl') {
         deltaX *= -1;
       }
       // if moving up/down only, no need to resize
-      if (deltaX !== 0) {
+      if (deltaX !== 0 || (pointerType === 'keyboard' && deltaY !== 0)) {
         if (pointerType === 'keyboard') {
+          if (deltaY !== 0 && deltaX === 0) {
+            deltaX = deltaY * -1;
+          }
           deltaX *= 10;
         }
```

Comment:
bit of a nitpick, but could you pull out the keyboard specific logic out of this if? I found it a bit confusing to think through what this logic was covering at first glance, but it feels like you could do 
```
if (pointerType === 'keyboard') {
   // do calculations to determine if up/down or left/right and adjust deltaX value accordingly
}

if (deltaX !== 0 ) {
   // rest of the old logic goes here
}
``